www/js/history/history-factory.js

```diff
@@ -70,6 +70,18 @@ angular.module('starter.history-factory', ['ionic'])
   };
 
   /**
+   * removeFromHistory()
+   * @description: Removes a suggestion from the histories array.
+   * @param suggestion A suggestion object
+   * @returns None
+   */
+  history.removeFromHistory = function(suggestion) {
+    if (history.isInHistories(suggestion)) { 
+      history.histories.splice(0, 1);
```

Comment:
this is just removing the first history object in the array. you should be searching through the array to find the exact location of the history object to remove and splice it out of that index.